src/Interpreters/Session.cpp

```diff
@@ -420,30 +441,31 @@ ContextMutablePtr Session::makeQueryContext(ClientInfo && query_client_info) con
 
 std::shared_ptr<SessionLog> Session::getSessionLog() const
 {
-    /// For the LOCAL interface we don't send events to the session log
-    /// because the LOCAL interface is internal, it does nothing with networking.
-    if (interface == ClientInfo::Interface::LOCAL)
-        return nullptr;
-
     // take it from global context, since it outlives the Session and always available.
     // please note that server may have session_log disabled, hence this may return nullptr.
     return global_context->getSessionLog();
 }
 
 ContextMutablePtr Session::makeQueryContextImpl(const ClientInfo * client_info_to_copy, ClientInfo * client_info_to_move) const
 {
+    if (!user_id && !is_internal_interserver_query)
```

Comment:
Can we check ClientInfo.interface instead? The function `authenticateInterserverFake()` looks pretty bad.

Reply:
This flag makes assertion more strict and I thought that `authenticateInterserverFake()` just makes thing more clear, but yes, checking `client_info.interface` will look better.